src/tools/compiletest/src/header.rs

```diff
@@ -642,14 +644,42 @@ impl Config {
     /// Parses a name-value directive which contains config-specific information, e.g., `ignore-x86`
     /// or `normalize-stderr-32bit`.
     fn parse_cfg_name_directive(&self, line: &str, prefix: &str) -> ParsedNameDirective {
+        // This matches optional whitespace, followed by a group containing a series of word
+        // characters (including '_' and '-'), followed optionally by a sequence consisting
+        // of a colon, optional whitespace, and another group containing word characters.
+        //
+        // Matches in full:
+        //   cfg-target-has-atomic: 128
+        //   cfg-target-has-atomic:128
+        //   cfg-target-has-atomic
```

Comment:
cfg: target-has-atomic=128 rather than this syntax might be a bit better - it matches more closely the syntax used in cfg and similar, which seems positive (and more easily extendable to any or other directives in the future).

Reply:
We could I guess? The reason I'm hesitating is that this syntax is standard for compiletest header directives. `=` would require you to remember that this particular directive works differently from all other directives, including other `only` and `ignore` directives, which are otherwise the same. That seems more confusing than breaking with the cfg format. This is also why I've used dashes, rather than underscores, for the examples.

Reply:
None of the other directives are trying to emulate cfg() attributes essentially; they have their own ad-hoc syntaxes. I think in this case it's closer to "this directive matches the stuff you can put inside cfg (with some limitations currently)".

Reply:
Oh, wait, I totally misread your comment. Yeah, that... would work fine, actually. I can do that.